x-pack/plugins/apm/public/store/urlParams.ts

```diff
@@ -138,8 +138,13 @@ function getPathParams(pathname: string) {
   }
 }
 
+interface TimeUpdate {
+  min: string;
+  max: string;
+}
+
 // ACTION CREATORS
-export function updateTimePicker(time: string) {
+export function updateTimePicker(time: TimeUpdate) {
   return { type: TIMEPICKER_UPDATE, time };
 }
```

Comment:
We can get rid of redux entirely when updating the time. `onTimeChange` should simply change the url.

Reply:
Hmm, I think we still need to dispatch with the datemath-parsed values so they can be assigned to `urlParams.start` and `urlParams.end` 

https://github.com/elastic/kibana/blob/a0bc7dae55694d99873e511fb8e0ac0bbd419338/x-pack/plugins/apm/public/store/urlParams.ts#L80

Or else we need to do the datemath parsing inside the location update reducer, I think?

Reply:
Oh also: during auto-refresh, we often don't need to update the URL (because the URL is a relative date string) but we will need to recalculate the parsed datemath values and update them in the store, so I think the TIMEPICKER_UPDATE is still necessary.

Reply:
> Oh also: during auto-refresh, we often don't need to update the URL (because the URL is a relative date string) but we will need to recalculate the parsed datemath values and update them in the store, so I think the TIMEPICKER_UPDATE is still necessary.

Good point - didn't think of that.